homeassistant/components/sabnzbd/sensor.py

```diff
@@ -1,39 +1,114 @@
 """Support for monitoring an SABnzbd NZB client."""
 from __future__ import annotations
 
-from homeassistant.components.sensor import SensorEntity
-from homeassistant.core import HomeAssistant
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from . import (
-    DATA_SABNZBD,
-    SENSOR_TYPES,
-    SIGNAL_SABNZBD_UPDATED,
-    SabnzbdSensorEntityDescription,
+
+from . import DOMAIN, SIGNAL_SABNZBD_UPDATED, SabnzbdApiData
+from ...config_entries import ConfigEntry
+from ...const import DATA_GIGABYTES, DATA_MEGABYTES, DATA_RATE_MEGABYTES_PER_SECOND
+from ...core import HomeAssistant
+from ...helpers.entity_platform import AddEntitiesCallback
+from .const import KEY_API, KEY_NAME
+
+
+@dataclass
+class SabnzbdRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    field_name: str
+
+
+@dataclass
+class SabnzbdSensorEntityDescription(SensorEntityDescription, SabnzbdRequiredKeysMixin):
+    """Describes Sabnzbd sensor entity."""
+
+
+SENSOR_TYPES: tuple[SabnzbdSensorEntityDescription, ...] = (
+    SabnzbdSensorEntityDescription(
+        key="current_status",
+        name="Status",
+        field_name="status",
```

Comment:
What is the difference between `key` and `field_name`  ?

Reply:
`field_name` is internal to sabnzbd api and `key` is for the component

Reply:
Why not align them?

Reply:
I can't change the name in `field_name` as it's an API of sabnzbd. Some of the `field name` have unreadable names that are not user friendly so hence the `key` field.

Reply:
But the `name` field is already there to be readable? The entity ID is also based on the `name` field.

Reply:
right. Aligned them.
Thanks.